compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -352,46 +353,46 @@ object CheckUnused:
    * - usage
    */
   private class UnusedData:
-    import collection.mutable.{Set => MutSet, Map => MutMap, Stack => MutStack, ListBuffer => MutList}
+    import collection.mutable as mut, mut.Stack, mut.ListBuffer
     import UnusedData.*
 
     /** The current scope during the tree traversal */
-    val currScopeType: MutStack[ScopeType] = MutStack(ScopeType.Other)
+    val currScopeType: Stack[ScopeType] = Stack(ScopeType.Other)
 
     var unusedAggregate: Option[UnusedResult] = None
 
     /* IMPORTS */
-    private val impInScope = MutStack(MutList[ImportSelectorData]())
+    private val impInScope = Stack(ListBuffer.empty[ImportSelectorData])
     /**
      * We store the symbol along with their accessibility without import.
      * Accessibility to their definition in outer context/scope
      *
      * See the `isAccessibleAsIdent` extension method below in the file
      */
-    private val usedInScope = MutStack(MutSet[(Symbol, Option[Name], Boolean)]())
-    private val usedInPosition = MutMap.empty[Name, MutSet[Symbol]]
+    private val usedInScope = Stack(mut.Set.empty[(Symbol, Option[Name], Type, Boolean)])
```

Comment:
This is suspicious. `Type`s don't have a meaningful `==` test in general. Why is it OK to use a `Type` in a `Set` here?